### ISSUE TITLE
gdPutBuf return value check for GIF

### ISSUE THREAD

**Issue by meweez** [libgd/libgd#757]
Partial fix applied for BMP and WebP. For GIF, a lot of refactoring is needed and I am not sure it is worth it. What do you think?

We can open a separate issue for GIF as it also requires ABI/API breaks for what I see.

_Originally posted by @pierrejoye in https://github.com/libgd/libgd/issues/750#issuecomment-914872385_

I think it is worth it because in function `gdImageGifAnimAdd` the first argument can be crafted and it is propagated into gdPutbuf.

**Comment by pierrejoye** [comment]
yes, however it can't be done in a useful manner without refactoring most
of the calls (up to the exposed functions). That can only happen in 2.4 or
later.

**Comment by pierrejoye** [comment]
It could be for 2.4 but I am not sure I will have the time to do it tho' :)
